torch/utils/data/datapipes/iter/ziparchivereader.py

```diff
@@ -25,42 +24,36 @@ class ZipArchiveReaderIterDataPipe(IterDataPipe[Tuple[str, BufferedIOBase]]):
     """
     def __init__(
             self,
-            datapipe: Iterable[Tuple[str, BufferedIOBase]],
+            datapipe: Iterable[str],
             length: int = -1):
         super().__init__()
-        self.datapipe: Iterable[Tuple[str, BufferedIOBase]] = datapipe
+        self.datapipe: Iterable[str] = datapipe
         self.length: int = length
 
     def __iter__(self) -> Iterator[Tuple[str, BufferedIOBase]]:
         if not isinstance(self.datapipe, Iterable):
-            raise TypeError("datapipe must be Iterable type but got {}".format(type(self.datapipe)))
-        for data in self.datapipe:
-            validate_pathname_binary_tuple(data)
-            pathname, data_stream = data
+            raise TypeError(f"datapipe must be Iterable type but got {type(self.datapipe)}")
+        for pathname in self.datapipe:
+            if not isinstance(pathname, str):
+                raise TypeError(f"pathname should be of string type, but is type {type(pathname)}")
             try:
-                # typing.cast is used here to silence mypy's type checker
-                zips = zipfile.ZipFile(cast(IO[bytes], data_stream))
+                zips = zipfile.ZipFile(pathname)
                 for zipinfo in zips.infolist():
                     # major version should always be 3 here.
                     if sys.version_info[1] >= 6:
                         if zipinfo.is_dir():
                             continue
                     elif zipinfo.filename.endswith('/'):
                         continue
-
                     extracted_fobj = zips.open(zipinfo)
                     inner_pathname = os.path.normpath(os.path.join(pathname, zipinfo.filename))
-                    # Add a reference of the source zipfile into extracted_fobj, so the source
-                    # zipfile handle won't be released until all the extracted file objs are destroyed.
-                    extracted_fobj.source_ref = zips  # type: ignore[attr-defined]
```

Comment:
I can't come up with a test case to show that this is still necessary. Let me know if you can think of something.